pipelines/base/config-dump.yaml

```diff
@@ -0,0 +1,114 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: dump-config
+spec:
+  params:
+  - name: REPO_NAME
+    type: string
+    default: .github
+  - name: BRANCH_NAME
+    type: string
+    default: add-initial-peribolos-config
+  - name: BASE_BRANCH
+    type: string
+    default: main
+  - name: GIT_EMAIL
+    type: string
+  - name: GIT_NAME
+    type: string
+  workspaces:
+  - name: secret-data
+    description: location for oauth token and ssh config
+  steps:
+    - name: dump-and-push-config
+      image: quay.io/operate-first/opf-toolbox
+      script: |
+        #!/bin/bash
+
+        TOKEN=$( cat $(workspaces.secret-data.path)/token )
+        ORG_NAME=$( cat $(workspaces.secret-data.path)/orgName )
+
+        git config --global user.email "$(params.GIT_EMAIL)"
+        git config --global user.name "$(params.GIT_NAME)"
+        
+        mkdir repo
+        cd repo
+        git clone https://$TOKEN@github.com/$ORG_NAME/$(params.REPO_NAME) .
+      
+        # Check if repo is empty
+        if [ $( git count-objects | awk '{ print $1 }' ) -eq "0" ]; then
+          echo "Repository empty, creating a README on $(params.BASE_BRANCH)"
+          echo "# .github" > README.md
+          git add .
+          git commit -m "Initial commit"
+        fi
+        
+        echo "Dumping org config"
+        peribolos --dump $ORG_NAME --github-token-path $(workspaces.secret-data.path)/token > config.yaml
+        
+        echo "Commiting config to $(params.BRANCH_NAME) branch"
+        git add .
+
+        # Check if branch is already created
+        git show-branch origin/$(params.BRANCH_NAME)
+        if [ $? -eq 0 ]; then
+          echo "Branch already exists"
+          git checkout $(params.BRANCH_NAME)  
+        else
+          echo "Branch doesn't exists, creating branch $(params.BRANCH_NAME)"
+          git checkout -b $(params.BRANCH_NAME)
+        fi
+        
+        git commit -m "Add initial yaml config"
+        git push --all origin
```

Comment:
The git flow here seems to be a bit odd.

1. You check if the repo is empty (what is the aim here? To check if it's a newly created repository? In that case we won't be able to clone it, since you can't clone a repo without commits).
2. If the repo is empty you create a README and a commit in the default branch.
3. You run peribolos and get the `config.yaml` file as an result.
4. You stage the `config.yaml`
5. You check if there's a branch `add-initial-peribolos-config` present. If yes, you checkout to it, if not you create branch and checkout to it. I sense a problem here. If you already have that branch present in origin and you try checkout to it, you get a conflict, because you have staged `config.yaml` from different history.
6. You commit and push to default branch as well as to the `add-initial-peribolos-config` branch.

I think some steps should be reversed/changed:

1. If it's a bare repository without commits, the `git clone` will fail. In such case you should do `git init` and `git remote add`.
2. You don't have to create a stub README (though we may want to, if we have meaningful content, describing peribolos and `.github` repositories in general.) You can always create an empty commit by `git commit --allow-empty`
3. You should push this empty commit to the default branch 
4. Now you should checkout the `add-initial-peribolos-config` branch (create if not present)
5. Run peribolos, since you are already on the branch you're not introducing conflicts.
6. Stage `config.yaml` and commit and push.
7. Open PR.